.buildkite/pull_request_pipeline.yml

```
@@ -123,6 +123,25 @@ steps:
       manual:
         allowed: true
 
+  - label: "Observability SRE container smoke test"
+    key: "observability-sre-container-smoke-test"
+    agents:
+      provider: gcp
+      imageProject: elastic-images-prod
+      image: family/platform-ingest-logstash-ubuntu-2204
+      machineType: "n2-standard-4"
+      diskSizeGb: 64
+    retry:
+      automatic:
+        - limit: 3
+    command: |
+      set -euo pipefail
+      source .buildkite/scripts/common/vm-agent.sh
+      ./gradlew --stacktrace artifactDockerObservabilitySRE
+      logstash_version="$(awk -F':' '{ if ("logstash" == $1) { gsub(/ /,"",$2); print $2; exit } }' versions.yml)-SNAPSHOT"
+      docker run docker.elastic.co/logstash/logstash-observability-sre:${logstash_version} \
```

It looks like `$` can be doubled to escape from buildkite interpolation: https://buildkite.com/docs/pipelines/configure/environment-variables#runtime-variable-interpolation

```suggestion
      logstash_version="$(awk -F':' '{ if ("logstash" == $$1) { gsub(/ /,"",$$2); print $$2; exit } }' versions.yml)-SNAPSHOT"
      docker run docker.elastic.co/logstash/logstash-observability-sre:$${logstash_version} \
```

---

I find bash already hard enough to read, thinking about buildkite interpolation on top of it makes it even more difficult. I would preper to extract into a shell script if for no other reason than readability.

---

Oh, sweet. I'll refactor to just source this and use the env var. This is great!

---

Oh, oops I didnt see https://github.com/elastic/logstash/pull/17311

---

I broke off 2fb809abbac931934a31299b4d6360a7fbe8aa6b so it can be cherry-picked over here (and resolved bugs, like the bash version of `printf` not being happy).